parser.y

```diff
@@ -11109,6 +11109,18 @@ StatementList:
 			if lexer, ok := yylex.(stmtTexter); ok {
 				s.SetText(parser.lexer.client, lexer.stmtText())
 			}
+
+			// fill arana hints
+			if hints := parser.lexer.aranaHints; len(hints) > 0 {
+				type hintSetter interface {
```

Comment:
What do you think about this way of writing (suggestions only)?
```yacc
import (
	"strings"

	"github.com/arana-db/parser/mysql"
	"github.com/arana-db/parser/ast"
	"github.com/arana-db/parser/model"
	"github.com/arana-db/parser/opcode"
	"github.com/arana-db/parser/auth"
	"github.com/arana-db/parser/charset"
	"github.com/arana-db/parser/types"
)

type hintSetter interface {
		SetHints(hints []string)
}

%}
```

Reply:
👍